### PR TITLE
feat(core): detect package manager used to invoke create-nx-(plugin|workspace)

### PR DESCRIPTION
**Pull request by nartc**
## Current Behavior

Currently you always have to specific —packageManager with create-nx-workspace. yarn create nx-workspace should use yarn as the package manager by default. when using pnpx, we should use pnpm.

## Expected Behavior

When consumers create a new Workspace (or Plugin) using the `create-nx-workspace` (or `create-nx-plugin`) generator, the package manager used to invoke the generator will be detected and used as `packageManager`. For example: `pnpx create-nx-workspace` will use `pnpm`, `yarn create nx-workspace` will use `yarn`. Explicit `--packageManager` flag will be priority over the detection.

## Related Issue(s)

Fixes #

### CODE REVIEW

File: e2e/workspace/src/create-nx-workspace.test.ts [hunk 1]
```diff
@@ -242,4 +242,40 @@ describe('create-nx-workspace', () => {
...
+      checkFilesDoNotExist(`yarn.lock`, `pnpm-lock.yam`);
+    }, 90000);
+
+    it('should use pnpm when invoked with pnpx', () => {
+      setupProject('pnpm');
+      checkFilesExist(`pnpm-lock.yaml`);
+      checkFilesDoNotExist(`yarn.lock`, `package-lock.json`);
+    }, 90000);
+
+    // skipping due to packageManagerCommand for createWorkspace not using yarn create nx-workspace
```
Comment (FrozenPandaz) [inline e2e/workspace/src/create-nx-workspace.test.ts:274]: Any idea why it doesn't?

File: e2e/utils/index.ts [hunk 2]
```diff
@@ -100,7 +103,7 @@ export function runCreateWorkspace(
     command += ` --defaultBase="${base}"`;
   }
 
-  if (packageManager) {
+  if (packageManager && !useDetectedPm) {
```
Comment (FrozenPandaz) [inline e2e/utils/index.ts:106]: Isn't `useDetectedPm = false` the same as `packageManger = null`? Do we need an extra flag here?

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: packages/tao/src/shared/package-manager.ts [hunk 3]
```diff
@@ -86,3 +86,32 @@ export function getPackageManagerVersion(
...
+/**
+ * Detects which package manager was used to invoke create-nx-{plugin|workspace} command
+ * based on the main Module process that invokes the command
+ * - npx returns 'npm'
+ * - pnpx returns 'pnpm'
+ * - yarn create returns 'yarn'
+ *
+ * Default to 'npm'
+ */
+export function detectInvokedPackageManager(): PackageManager {
```
[the remark this answers is not in the document]
Comment (nartc) [reply packages/tao/src/shared/package-manager.ts:99]: `create-nx-plugin` seems to use `package-manager` from `tao`.

File: e2e/workspace/src/create-nx-workspace.test.ts (hunk 1 above)
Comment (nartc) [reply e2e/workspace/src/create-nx-workspace.test.ts:272]: image
There's this comment in `getPackageManagerCommand`.

File: e2e/utils/index.ts (hunk 2 above)
Comment (nartc) [reply e2e/utils/index.ts:106]: `useDetectedPm` is a flag to **NOT** set `--package-manager`. This is purely for testing purpose. `packageManager` seems to be set (in test env) via `SELECTED_PM` env variable, and the package commands are fetched from that selected package manager (and is default to `npm`). For the purpose of testing the detection, I'd like to **NOT** set `--package-manager` when there is `packageManager` defined, but still able to have  `getPackageManagerCommand` run correctly.

**Review by AgentEnder — APPROVED** [review]
LGTM, wait for @FrozenPandaz though 😄.

**Comment by nartc** [comment]
Messed up a rebase on this one. Please check this one out https://github.com/nrwl/nx/pull/6749

**Pull request closed without merging**